### PR TITLE
refactor: complete migration of end-to-end.rs test to use #[tokio::test]

### LINKED ISSUE #890: Improved end-to-end testing
The current end-to-end testing performed by `tests/end-to-end.rs` is not ideal.

In particular:

* The log output is quite hard to parse
* It isn't possible to run specific tests individually
* All tests are reported as a single success/failure

I'm not entirely sure what the best approach here is, and we probably want to wait until we have a more comprehensive set of such tests, but creating an issue to track.

There is unstable support for [custom test frameworks](https://doc.rust-lang.org/unstable-book/language-features/custom-test-frameworks.html), but it may be necessary to implement something truly custom - it's unclear to me at this stage.

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
Closes #890 

I want it to be  quick and efficient to write new gRPC API and CLI tests (which is painful enough as it is)

# Changes

* Pull `Scenario` out into its own module, and encapsulate more functionality there
* Randomly create org_id and bucket_ids as part of `Scenario` creation so tests can be run in parallel
* Remove the last vestigates of a single long function and instead use `#[tokio::test]` everywhere

# Example output

Running the end to end tests is now a thing of beauty (if you are into that sort of thing)

```
-*- mode: compilation; default-directory: "~/Software/influxdb_iox/" -*-
Compilation started at Fri Mar 12 18:16:08

cd /Users/alamb/Software/influxdb_iox && cargo test --test end-to-end -- 
    Finished test [unoptimized + debuginfo] target(s) in 0.55s
     Running target/debug/deps/end_to_end-bfab96b4cb69e7af

running 30 tests
test end_to_end_cases::management_api::test_partition_list_error ... ok
test end_to_end_cases::management_api::test_set_get_writer_id ... ok
test end_to_end_cases::http::test_http_error_messages ... ok
test end_to_end_cases::management_api::test_create_database_invalid_name ... ok
test end_to_end_cases::management_api::test_create_get_database ... ok
test end_to_end_cases::management_api::test_list_databases ... ok
test end_to_end_cases::management_api::test_create_database_duplicate_name ... ok
test end_to_end_cases::management_api::test_partition_get ... ok
test end_to_end_cases::management_api::test_partition_list ... ok
test end_to_end_cases::management_api::test_chunk_get ... ok
test end_to_end_cases::management_api::test_partition_get_error ... ok
test end_to_end_cases::management_api::test_chunk_get_errors ... ok
test end_to_end_cases::flight_api::test ... ok
test end_to_end_cases::management_api::test_list_update_remotes ... ok
test end_to_end_cases::read_api::test ... ok
test end_to_end_cases::management_cli::test_list_chunks_error ... ok
test end_to_end_cases::management_cli::test_get_partition_error ... ok
test end_to_end_cases::storage_api::read_window_aggregate_test ... ok
test end_to_end_cases::management_cli::test_list_partitions_error ... ok
test end_to_end_cases::write_api::test_write ... ok
test end_to_end_cases::management_cli::test_get_chunks ... ok
test end_to_end_cases::management_cli::test_writer_id ... ok
test end_to_end_cases::storage_api::read_group_test ... ok
test end_to_end_cases::management_cli::test_list_partitions ... ok
test end_to_end_cases::management_cli::test_get_partition ... ok
test end_to_end_cases::storage_api::test ... ok
test end_to_end_cases::write_cli::test ... ok
test end_to_end_cases::management_cli::test_create_database ... ok
test end_to_end_cases::management_cli::test_remotes ... ok
test end_to_end_cases::read_cli::test ... ok

test result: ok. 30 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


Compilation finished at Fri Mar 12 18:16:10

```

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: tests/end-to-end.rs [hunk 1]
```diff
@@ -3,291 +3,7 @@
...
+// The tests are defined in the  submodules of end_to_end_cases
 
 pub mod common;
-
 mod end_to_end_cases;
-
-use common::server_fixture::*;
-
-#[tokio::test]
-async fn read_and_write_data() {
```
Comment (alamb) [inline tests/end-to-end.rs:33]: this common setup was moved into the individual test modules

File: tests/end-to-end.rs [hunk 2]
```diff
@@ -3,291 +3,7 @@
...
-        let expected_read_data = load_data(&influxdb2, &scenario).await;
-        let sql_query = "select * from cpu_load_short";
-
-        read_api::test(&fixture, &scenario, sql_query, &expected_read_data).await;
-        storage_api::test(&mut storage_client, &scenario).await;
-        flight_api::test(&fixture, &scenario, sql_query, &expected_read_data).await;
-    }
-}
-
-// TODO: Randomly generate org and bucket ids to ensure test data independence
```
Comment (alamb) [inline tests/end-to-end.rs:58]: TODO is done!

File: tests/end-to-end.rs [hunk 3]
```diff
@@ -3,291 +3,7 @@
...
-// where desired
-
-#[derive(Debug)]
-pub struct Scenario {
```
Comment (alamb) [inline tests/end-to-end.rs:62]: Moved into its own module

File: tests/end_to_end_cases/util.rs [hunk 4]
```diff
@@ -1,84 +0,0 @@
-use rand::{distributions::Alphanumeric, thread_rng, Rng};
```
Comment (alamb) [inline tests/end_to_end_cases/util.rs:1]: this code was moved into `scenario.rs`

File: tests/common/server_fixture.rs [hunk 5]
```diff
@@ -174,6 +174,24 @@ impl ServerFixture {
     pub fn influxdb2_client(&self) -> influxdb2_client::Client {
         influxdb2_client::Client::new(self.http_base(), TOKEN)
     }
+
+    /// Return a management client  suitable for communicating with this
```
Comment (tustvold) [inline tests/common/server_fixture.rs:178]: Extra space

File: tests/end-to-end.rs [hunk 6]
```diff
@@ -3,291 +3,7 @@
...
-use data_types::{names::org_and_bucket_to_database, DatabaseName};
-use end_to_end_cases::*;
-use generated_types::{
-    influxdata::iox::management::v1::DatabaseRules, storage_client::StorageClient, ReadSource,
-    TimestampRange,
-};
-
-type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
-type Result<T, E = Error> = std::result::Result<T, E>;
+// The tests are defined in the  submodules of end_to_end_cases
```
Comment (tustvold) [inline tests/end-to-end.rs:6]: Extra space

**Review by tustvold — APPROVED** [review]

**Comment by alamb** [comment]
Thanks for the review @tustvold !
